### ISSUE TITLE
enable_trailers: true not working?

### ISSUE THREAD

**Issue by 1liminal1** [custom-cards/upcoming-media-card#97]
After updating to latest 0.6.0 enable_trailers: true does not work and tries to open the deep links

## Describe the bug:
As above 

## Config YAML:
```
        elements:
          - type: custom:upcoming-media-card
            title: Recently Added
            label:
            date: ddmm
            style:
              top: 5px
              left: 5px
              width: 1px
              height: 260px
              transform: translate(0%, 0%)
            entity: sensor.plex_recently_added
            enable_trailers: true
            image_style: poster
            flag: false
            icon: none
            max: 8
            box_shadows: false
            border_color: transparent
            accent_color: transparent
            title_text: $empty
            line1_text: $day, $date
            line2_text: $number - $runtime
            line3_text: $empty
            line4_text: $empty
            title_size: large
            line1_size: large
            text_shadows: false
            card_mod:
              style: |
                ha-card.type-custom-upcoming-media-card {
                background: transparent;
                padding: 0px !important;
                box-shadow: none;
                overflow: visible !important;
                }
                ha-card > div {
                padding: 0px !important;
                }
                #main.plex_poster {
                margin: 0px !important;
                overflow: visible !important;
                position: absolute !important;
                width: 500px !important;
                }
                #main.plex_poster:nth-of-type(2) {
                left: 181px !important;
                }
                #main.plex_poster:nth-of-type(3) {
                left: 362px !important;
                top: 77px;
                }
                #main.plex_poster:nth-of-type(4) {
                top: 77px;
                left: 543px !important;
                }
                #main.plex_poster:nth-of-type(5) {
                top: 77px;
                left: 724px !important;
                }
                #main.plex_poster:nth-of-type(6) {
                top: 77px;
                left: 905px !important;
                }
                .plex_poster .plex_container_poster {
                border-radius: 10px;
                box-shadow: 5px 5px 20px rgba(0, 0, 0, 0.2);
                margin: 0px;
                padding: 0px;
                width: 168px;
                height: 258px;
                }
                .plex_poster .plex_svg_poster {
                margin: 0px;
                position: absolute;
                top: 166px;
                left: 0px;
                width: 153px !important;
                border-radius: 0px 10px 10px 0px;
                background-color: rgba(44, 45, 67, 0.6);
                backdrop-filter: blur(10px);
                -webkit-backdrop-filter: blur(10px);
                transition-duration: 150ms;
                }
                .plex_poster .plex_svg_poster:hover {
                opacity: 0%;
                transition-duration: 200ms;
                }
                .plex_line0_poster {
                font-size: 14px !important;
                }
                .plex_line1_poster {
                font-size: 21px !important;
                fill: white !important;
                opacity: 80%;
                }
                .plex_line2_poster {
                font-size: 22px !important;
                fill: white !important;
                opacity: 80%;
                }
```                

## Components:
Plex upcoming media (as above does nothing)
Sonarr tries to open sonarr
Radarr tries to open radarr

## HA Error:
No error

## Home Assistant Version:
2024.6.4

**Comment by mkanet** [comment]
Hi, did you remember to delete the cache in your browser after you restarted home assistant? It is required.

**Comment by 1liminal1** [comment]
That's still a thing? I cant do that, I need those cookies. 

I tried just removing files and that hasn't worked. Also explains why it doesn't work on the kiosk where I don't have access to things like cookies (which, even if I did, again I need for logins) 

It is strange though that trailers did work till I upgraded the latest release. 

Shame, anyway thanks

**Comment by 1liminal1** [comment]
So I tried just removing the HA cookie only and it changed nothing. Except of course i've lost all settings and need to set up the side bar and theme ect all again. Maybe you don't know how important cookies are these days, and I probably wouldn't have said anything if it worked... But it didn't. Please make sure you know what you're talking about before you ask people to just "clear their cache!!!!"

**Comment by mkanet** [comment]
It's not cookies you want to delete, its cached images and files.   The cached javascript file that doesn't go away.  I have to do this myself after each significant code change.  You can pretty much delete the entire cache; but keep your passwords and cookies.  Or, you could try on another web browser you never used with UMC.

I actually put this requirement right at the very top of the readme.MD
![image](https://github.com/custom-cards/upcoming-media-card/assets/811008/43cf498f-dec5-4d88-81a6-5246ce425b6e)

**Comment by 1liminal1** [comment]
That was the first thing I did and it didn't work. I then thought maybe you meant the cookie. My mistake. 

Steps I have taken

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/e7e1a6da-9391-496c-b6a3-b32d17fecb2d)

Clear the cache, restarted chrome.

Restarted HA

Click the link and tries to go to radarr or sonarr 

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/5d6706df-cf8b-47ff-b375-f272b4d74db6)

Ive also tried incognito and using another browser with the same results

**Comment by mkanet** [comment]
Something doesn't add up.  I have no idea why your plex_recently_added card (when clicking on items), would try to connect to your sonarr and radarr instance.

1. Are you trying UMC version 0.6.0 with plex_recently_added, radarr_upcoming_media. or sonarr_upcoming_media integrations (or all 3)?  Please also let me know which versions you have installed currently of each.

2.  Does clicking on deep_links, take you to the respective Plex, Radarr, Sonarr show or movie webpages when you remove  `enable_trailers: true` from the respective card codes? 

We just need to see what you're doing differently than others who are successfully using `enable_trailers: true`

**Comment by 1liminal1** [comment]
Ok I think we are misunderstanding again. Hopefully this will be clear.

- UMC is version 0.6.0
- radarr_upcoming_media 0.4.1 - Opens Radarr (GitHub is showing 0.4.1?)
- sonarr_upcoming_media 0.4.3 - Opens Sonarr 
- plex_recently_added 0.3.9 - Does nothing 

So it seems after the latest version UMC does not respect ```enable_trailers: true``` or ```diable_hyperlinks: true``` (i've tried both) it just defaults to vanilla options. 

I am running UI minimalist with a full YAML dashboard. I don't know what changed in 0.6.0 but the previous version I tried when you first told me about the trailers worked fine along with the disabled hyperlinks. 

In summary; this version does not respect key value for ```enable_trailers: true``` or ```diable_hyperlinks: true```

**Comment by mkanet** [comment]
I've only tested trailer playback on the vanilla home assistant install with Chrome, Edge web browsers and Home Assistant Companion for mobile devices (after enabling automatic video playback). Can you please see if it works for you on a basic Home Assistant lovelace dashboard to rule out compatibility issues?

It sound like you might have a similar problem the person below.. same exact symptoms, he got it working.
https://github.com/custom-cards/upcoming-media-card/issues/96#event-13273360623

Did you have video trailers working on the previous version of UMC? If so, can you revert to the previous version UMC 0.5.9 to see if it works again? Also, try uninstalling UMC completely, reboot home assistant, and install the latest version again. That might force it to install the newer version.

**Comment by 1liminal1** [comment]
Point 1, 

Thats what Im doing, I am not using Dakboard. I've tried two computers and its the same behaviour. 

Point 2

I've asked him what he means by "force clear lovelace cache"

Point 3

Ok, ill try that. It could be just something stuck on my side. But I just want to understand what that guy means by "force clear lovelace cache" first :)

**Comment by mkanet** [comment]
@1liminal1 "Force clear lovelace cache" is the same as clearing your web browser's cache; which you already did; plus.  You even tried in web browsers and also 2 other computers.  It's not something stuck in your web browser's cache; **it's stuck on your Home Assistant (server side) just like you suspected.**

Anyway, before you do anything, make sure you copy and paste the respective attribute data for your Radarr integration and post the code in your reply (Naviage to Home Assistant Developer Tools/STATES tab and type the word `Radarr` to find your `sensor.radarr_upcoming_media` sensor; then copy the first couple of items.  It should look something like this:

```
data:
  - title_default: $title
    line1_default: $episode
    line2_default: $release
    line3_default: $rating - $runtime
    line4_default: $number - $studio
    icon: mdi:arrow-down-bold
  - airdate: "2024-05-23T00:00:00Z"
    release: Available Online $day
    flag: true
    title: Civil War
    runtime: 109
    studio: DNA Films
    genres: War, Action, Drama
    rating: ""
    summary: >-
      In the near future, a group of war journalists attempt to survive while
      reporting the truth as the United States stands on the brink of civil war.
    trailer: https://www.youtube.com/watch?v=c2G18nIVpNE
    poster: https://image.tmdb.org/t/p/w500/sh7Rg8Er3tFcN9BpKIPOMvALgZd.jpg
    fanart: https://image.tmdb.org/t/p/w780/z121dSTR7PY9KxKuvwiIFSYW8cf.jpg
    deep_link: http://192.168.1.100:7878/movie/929590
  - airdate: "2024-06-06T00:00:00Z"
    release: Available Online $day
    flag: true
    title: Hit Man
    runtime: 116
    studio: Aggregate Films
    genres: Romance, Comedy, Crime
    rating: ""
    summary: >-
      A mild-mannered professor moonlighting as a fake hit man in police stings
      ignites a chain reaction of trouble when he falls for a potential client.
    trailer: https://www.youtube.com/watch?v=9a7C7Bxsm90
    poster: https://image.tmdb.org/t/p/w500/1126gjlBf4hTm9Sgf0ox3LGVEBt.jpg
    fanart: https://image.tmdb.org/t/p/w780/nv6F6tz7r61DUhE7zgHwLJFcTYp.jpg
    deep_link: http://192.168.1.100:7878/movie/974635
```
<br>

**Do you have either SSH or Samba addon setup on your Home Assistant instance (to have full access to your Home Assitant's `\config` folder?**  If so, do this:

<br>

**Clean install:**
1) Completely uninstall UMC from home assistant via HACS
2) Reboot Home Assistance
3)  After you reboot, navigate to the folder: `\\homeassistant\config\www\community\upcoming-media-card` (or wherever your Home Assistant instance installs HACS frontend cards).  Then, delete that folder.
4) After you confirm that this folder doesn't exist anymore.  Reboot.
5) Now... install cleanly version 0.6.0 from the `https://github.com/custom-cards/upcoming-media-card` repo.
6) **Delete your browser cache one last time.  Make sure you do this.**
7)  Now, try clicking on items in your Radarr, Sonarr, Plex UMC cards.  Make sure you still have the setting:  `enable_trailers: true` is still there for all other your cards.  Also make sure `disable_hyperlinks: true` isn't in the YAML code.
<br>
If it's not possible to access `\\homeassistant\config\www\community\upcoming-media-card` (or wherever you HACS installs your Home Assistant cards) do this:
<br>

**Normal uninstall/reinstall of UMC:**
1) Completely uninstall UMC from home assistant via HACS
2) Reboot Home Assistance
3) Install version 0.6.0 from the `https://github.com/custom-cards/upcoming-media-card` repo.
6) **Delete your browser cache one last time.  Make sure you do this.**
7)  Now, try clicking on items in your Radarr, Sonarr, Plex UMC cards.  Make sure you still have the setting:  `enable_trailers: true` in the respective YAML card code.  Also make sure `disable_hyperlinks: true` isn't there (since it will prevent cards from using trailers and deep_links).


**Please let me know which steps you followed above** (hopefully you were able to confirm that you deleted the upcoming-media-card install directory and its contents completely before installing again)

<br>

If the above methods don't work, uninstall UMC one more time.  **Reboot, and install upcoming_media_card version 0.5.8 from HACS**:
![image](https://github.com/custom-cards/upcoming-media-card/assets/811008/d953d6b6-cd0e-4fbf-b782-8940eab847ed)

**Comment by 1liminal1** [comment]
Hey sorry, its been a super busy few days. 

Here is the data from the radarr entity

Ill post the results of the clean install soon. 

```
data: 
- title_default: $title
  line1_default: $episode
  line2_default: $release
  line3_default: $rating - $runtime
  line4_default: $number - $studio
  icon: mdi:arrow-down-bold
- airdate: '2024-05-23T00:00:00Z'
  release: Available Online $day
  flag: true
  title: Civil War
  runtime: 109
  studio: DNA Films
  genres: War, Action, Drama
  rating: ''
  summary: >-
    In the near future, a group of war journalists attempt to survive while
    reporting the truth as the United States stands on the brink of civil war.
  trailer: https://www.youtube.com/watch?v=c2G18nIVpNE
  poster: https://image.tmdb.org/t/p/w500/sh7Rg8Er3tFcN9BpKIPOMvALgZd.jpg
  fanart: https://image.tmdb.org/t/p/w780/z121dSTR7PY9KxKuvwiIFSYW8cf.jpg
  deep_link: http://192.168.190.3:7878/movie/929590
- release: In Theaters $day, $date
  flag: false
  title: Deadpool & Wolverine
  runtime: 127
  studio: Marvel Studios
  genres: Science Fiction, Action, Comedy
  rating: ''
  summary: >-
    A listless Wade Wilson toils away in civilian life with his days as the
    morally flexible mercenary, Deadpool, behind him. But when his homeworld
    faces an existential threat, Wade must reluctantly suit-up again with an
    even more reluctant Wolverine.
  trailer: https://www.youtube.com/watch?v=73_1biulkYk
  poster: https://image.tmdb.org/t/p/w500/jbwYaoYWZwxtPP76AZnfYKQjCEB.jpg
  fanart: https://image.tmdb.org/t/p/w780/dvBCdCohwWbsP5qAaglOXagDMtk.jpg
  deep_link: http://192.168.190.3:7878/movie/533535
  ```

**Comment by 1liminal1** [comment]
Ok, so I can't seem to delete the card :( 

I tried the above and it was still the same, so I just removed the card and rebooted and all the elements are still there?

Everything is still here and the links go to Sonarr/ Radarr 

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/9c726662-2873-436c-84bd-504b4529f1e3)

But no card :( 

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/3304ff03-2920-47ff-b781-de022ed46d28)

I have to get on with other stuff, so no rush :) 

Thanks for all your help

**Comment by 1liminal1** [comment]
Got in to work this morning and it looks like its broken now. 

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/14e6f080-632b-4b99-b51b-fb07368000bc)

(I promise I deleted the cache lol) 

Ill reinstall today and let you know :)

**Comment by 1liminal1** [comment]
Just an FYI as well, there are two Upcoming Media Cards in HACS. But when looking at yours all the links are broken? It looks pretty dodgy HAHA. Anyway just to let you know :) 

![image](https://github.com/custom-cards/upcoming-media-card/assets/10505942/8de3add7-51b6-4c35-85f4-05b5a7a1acc0)

**Comment by 1liminal1** [comment]
Ok, 

Ive gone back to 0.5.8 and trailers are working again. Plex upcoming does nothing?

Thanks

**Comment by mkanet** [comment]
Hi @1liminal1 HACS doesn't display animated gifs.  You have to https://github.com/custom-cards/upcoming-media-card to see them in action.

**Comment by 1liminal1** [comment]
I have upgraded to 0.6.0 and it appears that trailers are now working consistently. Thank you. 

Yes, there seems to be multiple repos now both for the UMC and Plex recently added. 

I have removed the plex and have tried to re add, but getting errors saying this cannot be added from the UI :( 

Anyway, this is sorted. Thankyou, Ill open another ticket on plex recently added if I cant solve the issue.

**Comment by mkanet** [comment]
@1liminal1  If you're getting errors related to `cannot be added from the UI`, it means you're not using the original/correct repo(s) as I mentioned previously.  All original Home Assistant HACS repos all part of the original HACS repos **_custom-components_** for example:  
https://github.com/custom-components/sensor.plex_recently_added  
https://github.com/custom-components/sensor.radarr_upcoming_media
https://github.com/custom-components/sensor.sonarr_upcoming_media

Same with UMC; which is part of **_custom-cards_**  for example:  https://github.com/custom-cards/upcoming-media-card

It will be much easier to install these directly from HACS without having to manually add the repos once these repos are available on HACS again.  I submitted these PRs over 4 months ago!  It typically takes about 4 months for repos to be approved on HACS; so, it should be soon.  I'm not sure if all the HACS moderators are on summer vacation or what.

**Comment by 1liminal1** [comment]
Thanks for that, I got the right repo and now all trailers are working as they should :) 

Im sure other people will need this info as well, so thanks for posting. 

I didn't have to add the custom repos though... OR maybe I did it years ago when I first set this up. Regardless, all working. Thanks again!

**Comment by mkanet** [comment]
Youre welcome!  Honestly you're the first person who had so much trouble so far.  There are already many people already using these integrations and UMC.
